### ISSUE TITLE
Popen blocking on Windows

### ISSUE THREAD

**Issue by philippewarren** [arun11299/cpp-subprocess#75]
On Windows, the call to std::async in execute_process blocks until the process dies naturally, which is problematic if the process does not die and the intended use case is to kill it ourselves.
Could be easily fixed by storing the return value (of type std::future<void>) of the call to std::async as an instance member. This way, the wait on the future would only happen when the instance is destroyed.
This:
```cpp
  this->cleanup_future_ = std::async(std::launch::async, [this] {
    WaitForSingleObject(this->process_handle_, INFINITE);

    CloseHandle(this->stream_.g_hChildStd_ERR_Wr);
    CloseHandle(this->stream_.g_hChildStd_OUT_Wr);
    CloseHandle(this->stream_.g_hChildStd_IN_Rd);
  });
```
instead of this: 
```cpp
  std::async(std::launch::async, [this] {
    WaitForSingleObject(this->process_handle_, INFINITE);

    CloseHandle(this->stream_.g_hChildStd_ERR_Wr);
    CloseHandle(this->stream_.g_hChildStd_OUT_Wr);
    CloseHandle(this->stream_.g_hChildStd_IN_Rd);
  });
```

**Comment by HUMANIAM** [comment]
@philippewarren I faced the same issue on Windows when trying to create std::Popen with `subprocess::Popen(command, sp::output{sp::PIPE});`. I tried your suggested solution and it worked fine but what I don't understand is

- std::async with std::launch::async policy is supposed to run the lambda function in a new thread so the primary thread of the parent process shouldn't be blocked. Why is it blocked without std::future?
